src/test/javascript/spec/component/modeling-assessment-editor/modeling-assessment-editor.component.spec.ts

```diff
@@ -204,84 +214,85 @@ describe('ModelingAssessmentEditorComponent', () => {
-        component.modelingExercise.maxPoints = 10;
+    describe('save and submit', () => {
+        beforeEach(() => {
+            const course = new Course();
+            component.modelingExercise = new ModelingExercise(UMLDiagramType.ClassDiagram, course, undefined);
+            component.modelingExercise.assessmentDueDate = dayjs().subtract(2, 'days');
+            component.modelingExercise.maxPoints = 10;
 
-        const feedback = createTestFeedback();
-        component.unreferencedFeedback = [feedback];
+            const feedback = createTestFeedback();
+            component.unreferencedFeedback = [feedback];
 
-        component.result = {
-            id: 2374,
-            score: 8,
-            rated: true,
-            hasComplaint: false,
-        } as unknown as Result;
+            component.result = {
+                id: 2374,
+                score: 8,
+                rated: true,
+                hasComplaint: false,
+            } as unknown as Result;
 
-        component.submission = {
-            id: 1,
-            submitted: true,
-            type: 'MANUAL',
-            text: 'Test\n\nTest\n\nTest',
-        } as unknown as ModelingSubmission;
-        component.submission.results = [component.result];
-        getLatestSubmissionResult(component.submission)!.feedbacks = [
-            {
-                id: 2,
-                detailText: 'Feedback',
-                credits: 1,
-            } as Feedback,
-        ];
-        const saveAssessmentSpy = jest.spyOn(service, 'saveAssessment').mockReturnValue(of(getLatestSubmissionResult(component.submission)!));
+            component.submission = {
+                id: 1,
+                submitted: true,
+                type: 'MANUAL',
+                text: 'Test\n\nTest\n\nTest',
+            } as unknown as ModelingSubmission;
+            component.submission.results = [component.result];
+            getLatestSubmissionResult(component.submission)!.feedbacks = [
+                {
+                    id: 2,
+                    detailText: 'Feedback',
+                    credits: 1,
+                } as Feedback,
+            ];
+        });
 
-        component.ngOnInit();
-        tick(500);
-        component.onSaveAssessment();
-        expect(saveAssessmentSpy).toHaveBeenCalledOnce();
-    }));
+        it('should save assessment', fakeAsync(() => {
+            const saveAssessmentSpy = jest.spyOn(service, 'saveAssessment').mockReturnValue(of(getLatestSubmissionResult(component.submission)!));
 
-    it('should try to submit assessment', fakeAsync(() => {
-        const course = new Course();
-        component.modelingExercise = new ModelingExercise(UMLDiagramType.ClassDiagram, course, undefined);
-        component.modelingExercise.assessmentDueDate = dayjs().subtract(2, 'days');
+            component.ngOnInit();
+            tick(500);
```

Comment:
I think await component.ngOnInit() can be used.

Reply:
I even think it has to be used in order to avoid possible unwanted side-effects.

Reply:
Nevermind, the ngOnInit of the modeling assessment editor is not defined as async, so it can't be done here.